pgjdbc/src/main/java/org/postgresql/core/v3/CopyOperationImpl.java

```diff
@@ -49,9 +49,7 @@ public int getFormat() {
   }
 
   public boolean isActive() {
-    synchronized (castNonNull(queryExecutor)) {
-      return queryExecutor.hasLock(this);
-    }
+    return castNonNull(queryExecutor).hasLockOn(this);
   }
```

Comment:
This change adds the a new `hasLockOn(this)` method to QueryExecutorImpl which obtains the lock to run the hasLock() check. In this way the lock for QueryExecutorImpl isn't exposed.

An alternative approach would be to expose the QueryExecutorImpl lock via say an `obtainLock()` method and use it here in CopyOperationImpl like:
```java
try (ResourceLock queryExecutorLock = queryExecutor.obtainLock()) {  // or queryExecutor.lock.obtain()
  return queryExecutor.hasLock(this);
}
```